### PR TITLE
feat(cli): split `influxdb` feature properly

### LINKED ISSUE #865: Put `analytics` feature behind a CLI flag

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Linked Issues

* Closes #865

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### Config Changes
* [x] Ensure proper order in which CLI and config arguments are applied.
* [x] Ensure that config changes work with individual build features.

#### INX Changes
* [x] Run chronicle using an INX connection.
* [x] Use new `metrics_enabled` and `analytics_enabled` CLI/config flags

### CODE REVIEW

File: src/db/mod.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #[cfg(feature = "stardust")]
 pub mod collections;
 
-#[cfg(feature = "influxdb")]
+#[cfg(any(feature = "analytics", feature = "metrics"))]
 /// Module containing InfluxDb types and traits.
```
Comment (grtlr) [inline src/db/mod.rs:11]:
```suggestion
/// Module containing InfluxDb types and traits.
#[cfg(any(feature = "analytics", feature = "metrics"))]
```

**Review by grtlr — CHANGES_REQUESTED** [review]
Currently fails to compile with `cargo ci-check-features`.

**Review by grtlr — CHANGES_REQUESTED** [review]
The CLI flags don't work, because we never check for `metrics_enabled` in `stardust_inx`. We should probably add `metrics_enabled` and `analytics_enabled` fields to our `InfluxDb` type.

File: src/bin/inx-chronicle/stardust_inx/mod.rs [hunk 2]
```diff
@@ -239,7 +239,12 @@ impl InxWorker {
         stream: &mut (impl futures::Stream<Item = Result<LedgerUpdateMessage, InxError>> + Unpin),
     ) -> Result<(), InxWorkerError> {
         #[cfg(feature = "metrics")]
-        let start_time = self.influx_db.is_some().then(std::time::Instant::now);
+        let start_time = self
+            .influx_db
+            .as_ref()
+            .map(|db| db.config().metrics_enabled)
+            .unwrap_or_default()
+            .then(std::time::Instant::now);
```
[the remark this answers is not in the document]
Comment (Alexandcoats) [reply src/bin/inx-chronicle/stardust_inx/mod.rs:247]: Yeah, probably not a huge difference either way.

**Review by grtlr — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**